src/Our.Umbraco.DocTypeGridEditor/Web/UI/App_Plugins/DocTypeGridEditor/Js/doctypegrideditor.directives.js

```diff
@@ -0,0 +1,27 @@
+angular
+    .module('umbraco.directives')
+    .directive('compile', compile);
+
+compile.$inject = ['$compile'];
+
+function compile($compile) {
+    var directive = {
+        restrict: 'A',
+        scope: {
+            compile: '='
+        },
+        link: link
+    }
+
+    return directive;
+
+    function link($scope, $element, $attrs) {
+        $scope.$watch(function () { return $scope.compile }, function (newValue, oldValue) {
+            $element.html(newValue);
+
+            //Only compile child nodes to avoid starting an infinite loop
+            var childNodes = $element.contents();
+            $compile(childNodes)($scope);
+        });
+    }
+}
```

Comment:
Could you use a similar file structure as used in the other js files? ie, https://github.com/umco/umbraco-doc-type-grid-editor/blob/develop/src/Our.Umbraco.DocTypeGridEditor/Web/UI/App_Plugins/DocTypeGridEditor/Js/doctypegrideditor.controllers.js 

I find the $inject structure a bit confusing and the fact you are using the function before it's declared.

Reply:
Thanks for checking in, Matt.
I'll get it sorted out ASAP.